backend/tournesol/suggestions/suggester.py

```diff
@@ -0,0 +1,236 @@
+                )
+
+                self._complete_graph.add_node(
+                    self._entity_to_video[c.comparison.entity_2]
+                )
+
+            self._complete_graph.add_edge(
+                self._entity_to_video[c.comparison.entity_1],
+                self._entity_to_video[c.comparison.entity_2],
+            )
+
+        # create required user graphs (none at first in fact)
+
+    def _get_user_comparability_augmenting_videos(self) -> list[SuggestedVideo]:
+        """"
+        Function called to get the videos to recommend while the scale and translation
+        uncertainties of the user are not high enough
+        """
+        # I want all entities from the current poll compared by supertrusted user
+        # todo create alias to properly detect supertrusted ?
+        req_entities = Entity.objects \
+            .filter(Q(comparisons_entity_1__poll__name=self.poll.name) |
+                    Q(comparisons_entity_2__poll__name=self.poll.name))\
+            .filter(Q(comparisons_entity_1__user__is_staff=True) |
+                    Q(comparisons_entity_2__user__is_staff=True))\
+            .distinct()
+
+        return list(map(lambda entity: self._entity_to_video[entity], req_entities))
+
+    def _get_user_rate_later_video_list(self, user: User) -> list[SuggestedVideo]:
+        """
+        Function to get the list of videos of the user's rate later list
+        """
+        req_entities = Entity.objects \
+            .filter(type=self.poll.name) \
+            .filter(videoratelater__user__email=user.email)
+        return list(map(lambda entity: self._entity_to_video[entity], req_entities))
+
+    def do_offline_computation(self):
+        """
+        Function to call regularly to keep the variables useful for the suggestions up to date
+        with the collected information
+        """
+        # for requests => look at ml->inputs
+        scale_aug_vids = self._get_user_comparability_augmenting_videos()
+        for g in self._user_specific_graphs.values():
+            # Will be cached, do at registration
+            g.compute_offline_parameters(scale_aug_vids)
+
+    def register_new_user(self, new_user):
+        """
+        Function used to register a new user wanting suggestions, it thus initializes its
+        comparison graph
+        """
+        recommendation_user = RecommendationUser(
+            self._entity_to_video, new_user, self.criteria, self.poll
+        )
+        self._user_specific_graphs[new_user] = Graph(
+            local_user=recommendation_user,
+            local_poll=self.poll,
+            local_criteria=self.criteria,
+        )
+        # build user graph
+        query: QuerySet = (
+            ComparisonCriteriaScore.objects
+                                   .filter(comparison__poll__name=self.poll.name)
+                                   .filter(criteria=self.criteria)
+                                   .filter(comparison__user__email=new_user.email)
```

Comment:
Here tool, email should not be used, the user can be used directly. Django will use the primary key automatically.
```suggestion
                                   .filter(comparison__user=new_user)
```

Reply:
done thanks